lighthouse-core/test/audits/audit-test.js

```diff
@@ -99,4 +99,75 @@ describe('Audit', () => {
     assert.equal(result.score, null);
     assert.equal(result.scoreDisplayMode, 'error');
   });
+
+  describe('makeCodeSnippetDetails', () => {
+    const maxLinesAroundHighlight = 10;
+
+    it('Transforms code to lines array', () => {
+      const details = Audit.makeCodeSnippetDetails({
+        code: 'a\nb\nc',
+        title: 'Title',
+        highlights: [],
+      });
+
+      assert.equal(details.lines.length, 3);
+      assert.deepEqual(details.lines[1], {
+        number: 2,
+        content: 'b',
+      });
+    });
+
+    it('Truncates long lines', () => {
+      const details = Audit.makeCodeSnippetDetails({
+        code: Array(1001).join('-'),
+        title: 'Title',
+        highlights: [],
+      });
+
+      assert.equal(details.lines[0].truncated, true);
+      assert.ok(details.lines[0].content.length < 1000);
+    });
+
+    function makeLines(lineCount) {
+      return Array(lineCount + 1).join('-\n');
+    }
+
+    it('Limits the number of lines if there are no line-specific highlights', () => {
+      const details = Audit.makeCodeSnippetDetails({
+        code: makeLines(100),
+        title: 'Title',
+        highlights: [{
+          message: 'General',
+        }],
+        maxLinesAroundHighlight,
+      });
+      expect(details.lines.length).toBe(2 * maxLinesAroundHighlight + 1);
+    });
+
+    it('Limits the number of lines around highlights', () => {
+      const code = makeLines(99) + 'A\n' + makeLines(99) + '\nB';
+      const allLines = code.split('\n');
+      const details = Audit.makeCodeSnippetDetails({
+        code,
+        title: 'Title',
+        highlights: [{
+          lineNumber: allLines.findIndex(l => l === 'A') + 1,
```

Comment:
seems like we could probably just put `100` here :)

Reply:
I think this is more clear about what's being tested. `allLines.findIndex(...` -> "the line with A on it" is more immediately understandable than "100" -> "the line with A on it".